java/ql/lib/semmle/code/java/frameworks/apache/Collections.qll

```diff
@@ -182,6 +204,695 @@ private class ApacheKeyValueModel extends SummaryModelCsv {
+          ".list;PredicatedList;true;predicatedList;;;Element of Argument[0];Element of ReturnValue;value",
+          ".list;SetUniqueList;true;SetUniqueList;;;Element of Argument[0];Element of Argument[-1];value",
+          ".list;SetUniqueList;true;SetUniqueList;;;Element of Argument[1];Element of Argument[-1];value",
+          ".list;SetUniqueList;true;setUniqueList;;;Element of Argument[0];Element of ReturnValue;value",
+          ".list;SetUniqueList;true;asSet;;;Element of Argument[-1];Element of ReturnValue;value",
+          ".list;SetUniqueList;true;createSetBasedOnList;;;Element of Argument[1];Element of ReturnValue;value",
+          ".list;TransformedList;true;TransformedList;;;Element of Argument[0];Element of Argument[-1];value",
+          ".list;TransformedList;true;transformingList;;;Element of Argument[0];Element of ReturnValue;value",
+          ".list;TransformedList;true;getList;;;Element of Argument[-1];Element of ReturnValue;value",
+          ".list;TreeList;true;TreeList;;;Element of Argument[0];Element of Argument[-1];value",
+          ".list;UnmodifiableList;true;UnmodifiableList;;;Element of Argument[0];Element of Argument[-1];value",
+          ".list;UnmodifiableList;true;unmodifiableList;;;Element of Argument[0];Element of ReturnValue;value"
+        ]
+  }
+}
+
+/**
+ * Value-propagating models for the package `org.apache.commons.collections4.map`.
+ */
+private class ApacheMapModel extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      ["org.apache.commons.collections4", "org.apache.commons.collections"] +
+        [
+          // Note that when lambdas are supported we should have more models for DefaultedMap, LazyMap, TransformedMap, TransformedSortedMap
+          ".map;AbstractHashedMap;true;AbstractHashedMap;(Map);;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".map;AbstractHashedMap;true;AbstractHashedMap;(Map);;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".map;AbstractHashedMap;true;convertKey;;;Argument[0];ReturnValue;value",
+          ".map;AbstractHashedMap;true;entryKey;;;MapKey of Argument[-1];ReturnValue;value",
+          ".map;AbstractHashedMap;true;entryKey;;;MapKey of Argument[0];ReturnValue;value",
+          ".map;AbstractHashedMap;true;entryValue;;;MapValue of Argument[-1];ReturnValue;value",
+          ".map;AbstractHashedMap;true;entryValue;;;MapValue of Argument[0];ReturnValue;value",
+          ".map;AbstractHashedMap;true;createEntrySetIterator;;;MapKey of Element of ReturnValue; MapKey of Argument[-1];value",
+          ".map;AbstractHashedMap;true;createEntrySetIterator;;;MapValue of Element of ReturnValue; MapValue of Argument[-1];value",
+          ".map;AbstractHashedMap;true;createKeySetIterator;;;Element of ReturnValue; MapKey of Argument[-1];value",
+          ".map;AbstractHashedMap;true;createValuesIterator;;;Element of ReturnValue; MapValue of Argument[-1];value",
```

Comment:
Something looks odd here. Input has `ReturnValue`, output has spaces in it. And it seems like they're swapped